onchain/src/templates/component.cairo

```diff
@@ -59,11 +62,38 @@ pub mod TemplateStoreComponent {
             let template_id = self.templates_count.read();
             self.templates.write(template_id, template_metadata);
             self.templates_count.write(template_id + 1);
+            self
```

Comment:
Lets have this logic run only if `reward_token` and `reward` are non-zero, IE lets allow templates to be added with no reward.

Reply:
you mean IE lets NOT allow templates to be added with no reward right?

Reply:
I want to allow templates without rewards. Currently, if there is no reward token, the deposit function will revert with `Invalid token`.  So, `deposit` should only be called when `reward` and `reward_token` are set to non-zero values.